### ISSUE TITLE
Code duplicates

### ISSUE THREAD

**Issue by JonathanKuelz** [OstermD/EAIK#12]
The implementations of `IK_DH`, `IK_Homogeneous`, and `IK_URDF` [share a lot of common code](https://github.com/OstermD/EAIK/tree/main/src/eaik).

Is there any reason to not define an abstract robot base class, implement all their functionalities once, and inherit from this base class to define the modeling-specific functionalities?

That's something I can easily do as it's on the python side. However, I wanted to know whether you didn't do that intentionally @OstermD ?

**Comment by OstermD** [comment]
No, I initially thought of a completely different implementation but then stuck with the copy-paste code snippets.
This is indeed just some bad code design and should be added to a more concise class hierarchy.

**Comment by JonathanKuelz** [comment]
Did you already adapt the code? If not, feel free to reopen and assign me - I can refactor the code quickly

**Comment by OstermD** [comment]
Implemented in #19
